File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <bits/stdc++.h>
using namespace std;


bool solution(string s)
{
    bool answer = true;
    
    int count = 0;
    for (auto& item : s)
    {
        if (item == '(')
        {
            count++;
        }
        else if (item == ')')
        {
            count--;
        }
        
        if (count < 0)
        {
            answer = false;
            break;
        }
    }
    
    if (count != 0) answer = false;

    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True

    cnt=0
    if s[0]==')' or s[-1]=='(':
        answer=False
    else:
        for i in range(len(s)):
            if cnt<0:
                break
            elif s[i]=='(':
                cnt+=1
            elif s[i]==')':
                cnt-=1
    
    if cnt!=0:
        answer=False
    return answer